pkgs/development/python-modules/dlib/default.nix

```diff
@@ -20,17 +17,24 @@ buildPythonPackage {
       --replace "pytest==3.8" "pytest"
   '';
 
-  # although AVX can be enabled, we never test with it. Some Hydra machines
-  # fail because of this, however their build results are probably used on hardware
-  # with AVX support.
-  checkPhase = ''
-    ${python.interpreter} nix_run_setup test --no USE_AVX_INSTRUCTIONS
+  dontUseCmakeConfigure = true;
+
+  # Pass CMake flags through to the build script
+  preConfigure = ''
+    setupPyBuildFlags="$setupPyBuildFlags $(sed 's/-D\(\S*\)/--set \1/g' <<< "$cmakeFlags")"
```

Comment:
I'm ok with this, I guess. But to add more options:
- a `for`-loop would avoid spawning subprocesses, and `--set ${flag#-D}` might be more readable than `sed`
- one could also patch the upstream scripts e.g. like so: https://github.com/NixOS/nixpkgs/blob/b576bc2681ccc45a341f8dafa3361fe4e6bbfd1f/pkgs/development/python-modules/torchaudio/0001-setup.py-propagate-cmakeFlags.patch#L4

Reply:
The `sed` command also filters for `-D` prefixes, in case there are other items in `cmakeFlags`. I'll keep it as is for simplicity.

Reply:
I am not okay with this. Please use something simpler and easier.

Reply:
Done - now using a for loop.